### PR TITLE
Porting Analysis sample to Async

### PR DESCRIPTION
**Pull request by aji13187**
# Description

Ported 

## Type of change

- [ ] Bug fix
- [ ] New sample implementation
- [x] Sample viewer enhancement
- [ ] Other enhancement

**Platforms tested on**:

- [ ] Windows
- [ ] Android
- [ ] Linux
- [x] macOS
- [ ] iOS

## Checklist

- [ ] Runs and compiles on all active platforms as a standalone sample
- [x] Runs and compiles in the sample viewer(s)
- [x] Branch is up to date with the latest main/v.next
- [x] All merge conflicts have been resolved
- [x] Self-review of changes
- [x] There are no warnings related to changes
- [x] No unrelated changes have been made to any other code or project files
- [ ] Code is commented with correct formatting (CTRL+i)
- [x] All variable and method names are camel case
- [x] There is no leftover commented code
- [ ] Screenshots are correct size and display in description tab (500px by 500px, platform agnostic)
- [ ] If adding a new sample, it is added to the sample viewer
- [ ] Cherry-picked to Main branch (if applicable)

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/AnalyzeViewshed/AnalyzeViewshed.cpp [hunk 1]
```diff
@@ -162,65 +162,64 @@ void AnalyzeViewshed::calculateViewshed()
   // Assign a physical location to the new point feature based upon where the user clicked on the map view
   inputFeature->setGeometry(m_inputOverlay->graphics()->at(0)->geometry());
 
-  // connect to addFeature status changed signal
-  connect(inputFeatures, &FeatureCollectionTable::addFeatureCompleted, this, [this, inputFeatures](const QUuid&, bool success)
```
Comment (aji13187) [inline ArcGISRuntimeSDKQt_CppSamples/Analysis/AnalyzeViewshed/AnalyzeViewshed.cpp:166]: I have a question here, `addFeatureAsync()` does not return success/failure.
Is this intentional change in the API?
The TaskWatcher version of signal is this - (`void addFeatureCompleted(QUuid taskId, bool success);`)

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/Geotriggers/Geotriggers.cpp [hunk 2]
```diff
@@ -158,7 +158,8 @@ void Geotriggers::createGeotriggerMonitor(ServiceFeatureTable* serviceFeatureTab
   connect(geotriggerMonitor, &GeotriggerMonitor::geotriggerNotification, this, &Geotriggers::handleGeotriggerNotification);
 
   // Start must be explicitly called. It is called after the signal connection is defined to avoid a race condition in Qt.
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Analysis/Geotriggers/Geotriggers.cpp:160]: Review this comment. It does not sound valid anymore.

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/StatisticalQueryGroupSort/StatisticalQueryGroupSort.cpp [hunk 3]
```diff
@@ -93,43 +93,43 @@ void StatisticalQueryGroupSort::connectSignals()
...
-          sectionStrings << QString("\"%1\":\"%2\"").arg(it.key(), it.value().toString());
-      }
-      const QString sectionString = sectionStrings.join(',');
+    // get the group string
+    QStringList sectionStrings;
+    const QVariantMap& groupsMap = record->group();
+    for (auto it = groupsMap.cbegin(); it != groupsMap.cend(); ++it)
+    {
+        sectionStrings << QString("\"%1\":\"%2\"").arg(it.key(), it.value().toString());
+    }
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Analysis/StatisticalQueryGroupSort/StatisticalQueryGroupSort.cpp:122]:
```suggestion
    for (auto it = groupsMap.cbegin(); it != groupsMap.cend(); ++it)
    {
      sectionStrings << QString("\"%1\":\"%2\"").arg(it.key(), it.value().toString());
    }
```

This is old code, but let's fix it up while we're here.

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/AnalyzeViewshed/AnalyzeViewshed.cpp (hunk 1 above)
Comment (JamesMBallard) [reply ArcGISRuntimeSDKQt_CppSamples/Analysis/AnalyzeViewshed/AnalyzeViewshed.cpp:166]: That is the nice thing about the QFuture workflow. If you get into your .then handler, it succeeded.

For the bool==false, case you can hookup on `onFailed` handler if you want to handle that. In this case, it simply exits early. You don't need to do that anymore here.

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/Geotriggers/Geotriggers.cpp (hunk 2 above)
Comment (aji13187) [reply ArcGISRuntimeSDKQt_CppSamples/Analysis/Geotriggers/Geotriggers.cpp]: Modified the function name in comment - startAsync.

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/AnalyzeHotspots/AnalyzeHotspots.cpp [hunk 4]
```diff
@@ -149,7 +150,10 @@ void AnalyzeHotspots::processResults(GeoprocessingResult* result)
   connect(m_layer, &ArcGISMapImageLayer::doneLoading, this, [this](const Error& error)
   {
     if (error.isEmpty())
-        m_mapView->setViewpointGeometry(m_layer->fullExtent());
+    {
+      auto future = m_mapView->setViewpointGeometryAsync(m_layer->fullExtent());
+      Q_UNUSED(future);
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Analysis/AnalyzeHotspots/AnalyzeHotspots.cpp:155]: setViewpoint methods are no longer [[nodiscard]]. You shouldn't need Q_UNUSED or the future var anymore.

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/Geotriggers/Geotriggers.cpp [hunk 5]
```diff
@@ -157,8 +157,9 @@ void Geotriggers::createGeotriggerMonitor(ServiceFeatureTable* serviceFeatureTab
 
   connect(geotriggerMonitor, &GeotriggerMonitor::geotriggerNotification, this, &Geotriggers::handleGeotriggerNotification);
 
-  // Start must be explicitly called. It is called after the signal connection is defined to avoid a race condition in Qt.
-  geotriggerMonitor->start();
+  // startAsync must be explicitly called. It is called after the signal connection is defined to avoid a race condition in Qt.
+  auto future = geotriggerMonitor->startAsync();
+  Q_UNUSED(future);
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Analysis/Geotriggers/Geotriggers.cpp:162]:
```suggestion
  Q_UNUSED(future)
```

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/LineOfSightGeoElement/LineOfSightGeoElement.cpp [hunk 6]
```diff
@@ -229,7 +229,8 @@ void LineOfSightGeoElement::initialize()
     });
 
     Camera camera(observationPoint, 700, -30, 45, 0);
-    m_sceneView->setViewpointCamera(camera, 0);
+    auto future = m_sceneView->setViewpointCameraAsync(camera, 0);
+    Q_UNUSED(future);
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Analysis/LineOfSightGeoElement/LineOfSightGeoElement.cpp:233]: Ditto this is no longer [[nodiscard]]

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/LineOfSightLocation/LineOfSightLocation.cpp [hunk 7]
```diff
@@ -103,7 +104,8 @@ void LineOfSightLocation::setInitialViewpoint()
   const double pitch = 62;
   const double roll = 0;
   Camera camera(point, heading, pitch, roll);
-  m_sceneView->setViewpointCamera(camera);
+  auto future = m_sceneView->setViewpointCameraAsync(camera);
+  Q_UNUSED(future);
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/Analysis/LineOfSightLocation/LineOfSightLocation.cpp:108]: Ditto this is no longer [[nodiscard]]

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/AnalyzeHotspots/AnalyzeHotspots.cpp (hunk 4 above)
Comment (aji13187) [reply ArcGISRuntimeSDKQt_CppSamples/Analysis/AnalyzeHotspots/AnalyzeHotspots.cpp:155]: My Windows machine code was old, it confused me. Sorry!, I will revert it.

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/LineOfSightLocation/LineOfSightLocation.cpp (hunk 7 above)
Comment (aji13187) [reply ArcGISRuntimeSDKQt_CppSamples/Analysis/LineOfSightLocation/LineOfSightLocation.cpp]: Modified.

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/LineOfSightGeoElement/LineOfSightGeoElement.cpp (hunk 6 above)
Comment (aji13187) [reply ArcGISRuntimeSDKQt_CppSamples/Analysis/LineOfSightGeoElement/LineOfSightGeoElement.cpp]: Modified.

File: ArcGISRuntimeSDKQt_CppSamples/Analysis/AnalyzeHotspots/AnalyzeHotspots.cpp (hunk 4 above)
Comment (aji13187) [reply ArcGISRuntimeSDKQt_CppSamples/Analysis/AnalyzeHotspots/AnalyzeHotspots.cpp]: Done.

**Review by JamesMBallard — APPROVED** [review]

**Pull request merged**